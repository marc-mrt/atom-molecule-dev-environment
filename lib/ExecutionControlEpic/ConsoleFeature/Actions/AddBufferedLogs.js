"use babel";
// @flow

import type { ConsoleLog } from "../Types/types";

export type SetBufferedLogsForTaskAction = {
  type: "ADD_CONSOLE_LOGS",
  payload: {
    logs: Array<ConsoleLog>,
  },
};

export function addBufferedLogsForTask(
  logs: Array<ConsoleLog>,
): SetBufferedLogsForTaskAction {
  return {
    type: "ADD_CONSOLE_LOGS",
    payload: {
      logs,
    },
  };
}
