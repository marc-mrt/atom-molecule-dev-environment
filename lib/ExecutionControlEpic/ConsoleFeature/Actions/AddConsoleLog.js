"use babel";
// @flow

import type { ConsoleLog } from "../Types/types.js.flow";

export type SetConsoleLogsForTaskAction = {
  type: "ADD_CONSOLE_LOG",
  payload: {
    consoleLog: ConsoleLog,
  },
};

export function addConsoleLogsForTask(
  log: ConsoleLog,
): SetConsoleLogsForTaskAction {
  return {
    type: "ADD_CONSOLE_LOG",
    payload: {
      consoleLog: log,
    },
  };
}
