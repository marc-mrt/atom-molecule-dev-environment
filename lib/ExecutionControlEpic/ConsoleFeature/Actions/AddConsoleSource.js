"use babel";
// @flow

export type SetConsoleSource = {
  type: "ADD_CONSOLE_SOURCE",
  payload: {
    name: string,
    isChecked: boolean,
  },
};

export function addConsoleSource(
  source: string,
  status: boolean,
): SetConsoleSource {
  return {
    type: "ADD_CONSOLE_SOURCE",
    payload: {
      name: source,
      isChecked: status,
    },
  };
}
