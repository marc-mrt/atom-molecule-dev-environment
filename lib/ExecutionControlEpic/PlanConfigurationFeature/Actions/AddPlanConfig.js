"use babel";
// @flow

import type { DevToolInfo } from "../Types/types";
import type { PackageInfo } from "../../../ProjectSystemEpic/PackageFeature/Types/types";

export type AddPlanConfigAction = {
  type: "ADD_PLAN_CONFIGURATION",
  payload: {
    id: string,
    name: string,
    tool: DevToolInfo,
    config: mixed,
    packageInfo: PackageInfo,
  },
};

export function addPlanConfig(
  id: string,
  name: string,
  toolInfo: DevToolInfo,
  config: mixed,
  packageInfo: PackageInfo,
): AddPlanConfigAction {
  return {
    type: "ADD_PLAN_CONFIGURATION",
    payload: {
      id,
      name,
      tool: toolInfo,
      config,
      packageInfo,
    },
  };
}
