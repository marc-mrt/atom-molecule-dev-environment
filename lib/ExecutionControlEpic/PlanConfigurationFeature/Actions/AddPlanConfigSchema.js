"use babel";
// @flow

import type { ConfigSchemaPart, DevToolInfo } from "../Types/types";

export type AddPlanConfigSchemaAction = {
  type: "ADD_PLAN_CONFIGURATION_SCHEMA",
  payload: {
    tool: DevToolInfo,
    schema: ConfigSchemaPart,
  },
};

export function addPlanConfigSchema(
  tool: DevToolInfo,
  schema: ConfigSchemaPart,
): AddPlanConfigSchemaAction {
  return {
    type: "ADD_PLAN_CONFIGURATION_SCHEMA",
    payload: {
      schema,
      tool,
    },
  };
}
