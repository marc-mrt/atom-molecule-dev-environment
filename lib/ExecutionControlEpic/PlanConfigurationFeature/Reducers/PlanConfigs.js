"use babel";
// @flow

import type { PlanConfig } from "../Types/types.js.flow";
import { generatePlanId } from "../Model/id";
import { Map } from "immutable";

export default function(
  state: PlanConfigsReducer = Map(),
  action: any,
): PlanConfigsReducer {
  switch (action.type) {
    case "ADD_PLAN_CONFIGURATION":
      const id = generatePlanId();
      return state.setIn([action.payload.tool.id, id], {
        ...action.payload,
        id: id.toString(),
        pinned: false,
      });
    case "PIN_PLAN_CONFIGURATION":
      return state.updateIn(
        [action.payload.planConfig.tool.id, action.payload.planConfig.id],
        (plan = {}) => ({
          ...plan,
          pinned: true,
        }),
      );
    case "UNPIN_PLAN_CONFIGURATION":
      return state.updateIn(
        [action.payload.planConfig.tool.id, action.payload.planConfig.id],
        (plan = {}) => ({
          ...plan,
          pinned: false,
        }),
      );
    case "REMOVE_PLAN_CONFIGURATION":
      return state.deleteIn([
        action.payload.planConfig.tool.id,
        action.payload.planConfig.id,
      ]);
    default:
      return state;
  }
}

export type PlanConfigsReducer = Map<string, Map<string, PlanConfig>>;
