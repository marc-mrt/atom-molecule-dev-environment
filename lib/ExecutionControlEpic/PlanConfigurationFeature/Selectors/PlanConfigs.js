"use babel";
// @flow

import type { PlanConfig } from "../Types/types.js.flow";
import type { PlanConfigsReducer } from "../Reducers/PlanConfigs";
import { List } from "immutable";

export function selectPlansOfTool(
  state: PlanConfigsReducer,
  toolId: string,
): List<PlanConfig> {
  return state.get(toolId) ? state.get(toolId).toList() : List();
}

export function selectPinnedPlans(state: PlanConfigsReducer): List<PlanConfig> {
  return state.valueSeq().reduce((pinnedPlans, idMappedPlan) => {
    return pinnedPlans.concat(
      idMappedPlan.valueSeq().filter(plan => plan.pinned),
    );
  }, List());
}
