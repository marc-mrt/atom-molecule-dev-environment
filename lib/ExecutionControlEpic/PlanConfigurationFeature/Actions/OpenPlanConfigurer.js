"use babel";
// @flow

import type { DevToolInfo } from "../Types/types.js.flow";
import {
  renderToolConfigurer,
  renderToolPlans,
  toolPlansPanel,
} from "../AtomLinks/Panels";
import { sidePanel } from "../../../GlobalSystem/AtomLinks/Panels";
import { refreshAllProjectsPackages } from "../../../ProjectSystemEpic/PackageFeature/Actions/RefreshAllProjectsPackages";
import { DevToolsControllerInstance } from "../../../ExecutionControlEpic/DevtoolLoadingFeature/Model/DevToolsController";

export function openPlanConfigurer(tool: DevToolInfo) {
  return (dispatch: any) => {
    const plugins = DevToolsControllerInstance.getPackagesPlugins();
    renderToolConfigurer(sidePanel, tool);
    renderToolPlans(tool);
    toolPlansPanel.show();
    dispatch(refreshAllProjectsPackages(plugins));
  };
}
