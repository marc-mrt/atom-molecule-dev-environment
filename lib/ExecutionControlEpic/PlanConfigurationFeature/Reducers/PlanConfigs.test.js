"use babel";
// @flow

import PlanConfigs from "./PlanConfigs";
import type { AddPlanConfigAction } from "../Actions/AddPlanConfig";
import { addPlanConfig } from "../Actions/AddPlanConfig";
import type { PinPlanConfigAction } from "../Actions/PinPlanConfig";
import { pinPlanConfig } from "../Actions/PinPlanConfig";
import type { UnpinPlanConfigAction } from "../Actions/UnpinPlanConfig";
import { unpinPlanConfig } from "../Actions/UnpinPlanConfig";
import type { RemovePlanConfigAction } from "../Actions/RemovePlanConfig";
import { removePlanConfig } from "../Actions/RemovePlanConfig";
import type { PlanConfig } from "../Types/types.js.flow";
import { Map } from "immutable";

describe("PlanConfigs", () => {
  describe("ADD_PLAN_CONFIGURATION", () => {
    it("should add a plan configuration", () => {
      let state = Map();
      let action: AddPlanConfigAction = addPlanConfig(
        "0",
        "ls",
        {
          id: "toolid",
          name: "shell",
          iconUri: "atom://myplugin/icon.png",
          uri: "file:///file",
        },
        "ls -l",
        { type: "integrated" },
        {
          name: "",
          path: "/",
          type: "directory",
          uriPlatform: "posix",
        },
      );
      let subject = PlanConfigs(state, action);

      expect(subject).toMatchSnapshot();
    });
  });

  describe("PIN_PLAN_CONFIGURATION", () => {
    it("should pin plan config", () => {
      let planConfig: PlanConfig = {
        name: "ls",
        id: "0",
        tool: {
          id: "toolid",
          name: "shell",
          iconUri: "atom://myplugin/icon.png",
          uri: "file:///file",
        },
        config: "ls -l",
        stager: { type: "integrated" },
        packageInfo: {
          name: "",
          path: "/",
          type: "directory",
          uriPlatform: "posix",
        },
        pinned: false,
      };
      let planConfig2: PlanConfig = {
        name: "mimikatz",
        id: "1",
        tool: {
          id: "toolid",
          name: "powershell",
          iconUri: "atom://myplugin/icon.png",
          uri: "file:///file",
        },
        config: "Invoke-Mimikatz",
        stager: { type: "integrated" },
        packageInfo: {
          name: "",
          path: "/",
          type: "directory",
          uriPlatform: "posix",
        },
        pinned: false,
      };
      let state = Map([
        ["toolid", Map([["0", planConfig], ["1", planConfig2]])],
      ]);
      let action: PinPlanConfigAction = pinPlanConfig(
        Object.assign({}, planConfig, { state: null }),
      );
      let subject = PlanConfigs(state, action);

      expect(subject).toMatchSnapshot();
    });
  });

  describe("UNPIN_PLAN_CONFIGURATION", () => {
    it("should unpin plan config", () => {
      let planConfig: PlanConfig = {
        name: "ls",
        id: "0",
        tool: {
          id: "toolid",
          name: "shell",
          iconUri: "atom://myplugin/icon.png",
          uri: "file:///file",
        },
        config: "ls -l",
        stager: { type: "integrated" },
        packageInfo: {
          name: "",
          path: "/",
          type: "directory",
          uriPlatform: "posix",
        },
        pinned: true,
      };
      let planConfig2: PlanConfig = {
        name: "mimikatz",
        id: "1",
        tool: {
          id: "toolid",
          name: "powershell",
          iconUri: "atom://myplugin/icon.png",
          uri: "file:///file",
        },
        config: "Invoke-Mimikatz",
        stager: { type: "integrated" },
        packageInfo: {
          name: "",
          path: "/",
          type: "directory",
          uriPlatform: "posix",
        },
        pinned: false,
      };
      let state = Map([
        ["toolid", Map([["0", planConfig], ["1", planConfig2]])],
      ]);
      let action: UnpinPlanConfigAction = unpinPlanConfig(
        Object.assign({}, planConfig, { state: null }),
      );
      let subject = PlanConfigs(state, action);

      expect(subject).toMatchSnapshot();
    });
  });

  describe("REMOVE_PLAN_CONFIGURATION", () => {
    it("should remove plan config", () => {
      let planConfig: PlanConfig = {
        name: "ls",
        id: "0",
        tool: {
          id: "toolid",
          name: "shell",
          iconUri: "atom://myplugin/icon.png",
          uri: "file:///file",
        },
        config: "ls -l",
        stager: { type: "integrated" },
        packageInfo: {
          name: "",
          path: "/",
          type: "directory",
          uriPlatform: "posix",
        },
        pinned: false,
      };
      let planConfig2: PlanConfig = {
        name: "mimikatz",
        id: "1",
        tool: {
          id: "toolid",
          name: "powershell",
          iconUri: "atom://myplugin/icon.png",
          uri: "file:///file",
        },
        config: "Invoke-Mimikatz",
        stager: { type: "integrated" },
        packageInfo: {
          name: "",
          path: "/",
          type: "directory",
          uriPlatform: "posix",
        },
        pinned: false,
      };
      let state = Map([
        ["toolid", Map([["0", planConfig], ["1", planConfig2]])],
      ]);
      let action: RemovePlanConfigAction = removePlanConfig(
        Object.assign({}, planConfig, { state: null }),
      );
      let subject = PlanConfigs(state, action);

      expect(subject).toMatchSnapshot();
    });
  });
});
