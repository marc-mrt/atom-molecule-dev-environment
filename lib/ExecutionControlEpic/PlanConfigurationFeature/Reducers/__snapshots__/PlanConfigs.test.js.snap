// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanConfigs ADD_PLAN_CONFIGURATION should add a plan configuration 1`] = `
Immutable.Map {
  "toolid": Immutable.Map {
    "0": Object {
      "config": "ls -l",
      "id": "0",
      "name": "ls",
      "packageInfo": Object {
        "name": "",
        "path": "/",
        "type": "directory",
      },
      "pinned": false,
      "tool": Object {
        "iconUri": "atom://myplugin/icon.png",
        "id": "toolid",
        "name": "shell",
        "uri": "file:///file",
      },
    },
  },
}
`;

exports[`PlanConfigs PIN_PLAN_CONFIGURATION should pin plan config 1`] = `
Immutable.Map {
  "toolid": Immutable.Map {
    "0": Object {
      "config": "ls -l",
      "id": "0",
      "name": "ls",
      "packageInfo": Object {
        "name": "",
        "path": "/",
        "type": "directory",
      },
      "pinned": true,
      "tool": Object {
        "iconUri": "atom://myplugin/icon.png",
        "id": "toolid",
        "name": "shell",
        "uri": "file:///file",
      },
    },
    "1": Object {
      "config": "Invoke-Mimikatz",
      "id": "1",
      "name": "mimikatz",
      "packageInfo": Object {
        "name": "",
        "path": "/",
        "type": "directory",
      },
      "pinned": false,
      "tool": Object {
        "iconUri": "atom://myplugin/icon.png",
        "id": "toolid",
        "name": "powershell",
        "uri": "file:///file",
      },
    },
  },
}
`;

exports[`PlanConfigs REMOVE_PLAN_CONFIGURATION should remove plan config 1`] = `
Immutable.Map {
  "toolid": Immutable.Map {
    "1": Object {
      "config": "Invoke-Mimikatz",
      "id": "1",
      "name": "mimikatz",
      "packageInfo": Object {
        "name": "",
        "path": "/",
        "type": "directory",
      },
      "pinned": false,
      "tool": Object {
        "iconUri": "atom://myplugin/icon.png",
        "id": "toolid",
        "name": "powershell",
        "uri": "file:///file",
      },
    },
  },
}
`;

exports[`PlanConfigs UNPIN_PLAN_CONFIGURATION should unpin plan config 1`] = `
Immutable.Map {
  "toolid": Immutable.Map {
    "0": Object {
      "config": "ls -l",
      "id": "0",
      "name": "ls",
      "packageInfo": Object {
        "name": "",
        "path": "/",
        "type": "directory",
      },
      "pinned": false,
      "tool": Object {
        "iconUri": "atom://myplugin/icon.png",
        "id": "toolid",
        "name": "shell",
        "uri": "file:///file",
      },
    },
    "1": Object {
      "config": "Invoke-Mimikatz",
      "id": "1",
      "name": "mimikatz",
      "packageInfo": Object {
        "name": "",
        "path": "/",
        "type": "directory",
      },
      "pinned": false,
      "tool": Object {
        "iconUri": "atom://myplugin/icon.png",
        "id": "toolid",
        "name": "powershell",
        "uri": "file:///file",
      },
    },
  },
}
`;
